public/js/model/game.js

```diff
@@ -0,0 +1,145 @@
+function Game() {
+  this.rollHistory = {
+    1: [], 2: [] , 3: [], 4: [], 5: [], 6: [], 7: [], 8: [], 9: [], 10: [], 11:[]
+  };
+  this.scoreCard = {
+    1: "", 2: "", 3: "", 4: "", 5: "", 6: "", 7: "", 8: "", 9: "", 10: ""
+  };
+  this.cumulativeScore = {
+    1: "", 2: "", 3: "", 4: "", 5: "", 6: "", 7: "", 8: "", 9: "", 10: ""
+  };
+  this.insult = "Are you kidding me pandejo? Nobody f***s with the Jesus!"
+};
+
+Game.prototype.roll = function (score) {
+  for (var i = 1; i <= 11; i++) {
+    if (this._isNormalFrame(i)) {
+      if (this._isLegalRoll(i, score)) {
+        this.rollHistory[i].push(score);
+        this.insult = this._randomInsult()
+      } else {
+        this.insult = "You are a filthy cheater cabrón! Enter a valid score."
+      };
+      break;
+    } else if (this._isBonusRoll(i)) {
+      this.rollHistory[i].push(score);
+      break;
+    } else if (this._isBonusFrame(i)) {
+      this.rollHistory[i].push(score);
+      break;
+    };
+  };
+};
+
+Game.prototype.score = function () {
+  for (var i = 1; i <= 10; i++) {
+    if (this.rollHistory[i].length === 2) {
+      this.scoreCard[i] = this.rollHistory[i][0] + this.rollHistory[i][1];
+    } else if (this.rollHistory[i][0] === 10) {
+      this.scoreCard[i] = this.rollHistory[i][0];
+    };
+  };
+  for (var i = 1; i <= 9; i++) {
+    var nextTwoFrames = this.rollHistory[i + 1].concat(this.rollHistory[i + 2]);
+    if (this.rollHistory[i][0] === 10 && (nextTwoFrames[0] + nextTwoFrames[1])) {
+      this.scoreCard[i] += (nextTwoFrames[0] + nextTwoFrames[1]);
+    } else if (this.rollHistory[i][0] + this.rollHistory[i][1] === 10 && nextTwoFrames[0]) {
+      this.scoreCard[i] += nextTwoFrames[0]
+    };
+  };
+  if (this.rollHistory[10][0] === 10 && this.rollHistory[11][1]) {
+    this.scoreCard[10] += (this.rollHistory[11][0] + this.rollHistory[11][1]);
+  } else if (this.rollHistory[10][0] + this.rollHistory[10][1] === 10 && this.rollHistory[11][0]) {
+    this.scoreCard[10] += this.rollHistory[11][0];
+  };
+  return this.scoreCard
+};
+
+Game.prototype.accumulate = function () {
+  var scoreAccumulation = 0
+  for (var i = 1; i <= 10; i++) {
+    scoreAccumulation += this.scoreCard[i];
+    if (this.scoreCard[i] != "") {
+      this.cumulativeScore[i] = ' : ' + scoreAccumulation;
+    };
+  };
+  return this.cumulativeScore;
+};
+
+Game.prototype._isNormalFrame = function (frame) {
```

Comment:
I liked the way you have split out the responsibility of checking the frames: whether it is a normal frame, strike frame etc.